prql-compiler/src/parser/expr.rs

```diff
@@ -273,6 +274,38 @@ where
         .labelled("function call")
 }
 
+fn lambda_func() -> impl Parser<Token, FuncDef_, Error = Simple<Token>> {
+    (
+        // params
+        ident_part()
+            .then(type_expr().or_not())
+            .then(ctrl(':').ignore_then(expr()).or_not())
+            .repeated()
+    )
+    .then_ignore(just(Token::ArrowThin))
+    .then(expr_call().map(Box::new))
+    .then_ignore(new_line())
+    // .map(|(((name, return_ty), params), body)| {
+    .map(|(params, body)| {
+        let (pos, nam) = params
+            .into_iter()
+            .map(|((name, ty_expr), default_value)| FuncParam {
+                name,
+                ty_expr,
+                default_value,
+            })
```

Comment:
This part can be mapped earlier, right before `.repeated()`. Not sure if that's more readable, though.